### PR TITLE
Implement limited functionality for Split Pay - DirectDebitService::standaloneCredits

### PR DESCRIPTION
**Pull request by PayLeaseBrian**
This PR implements limited functionality relating to linked merchant accounts and split pay. Specifically it:
* enables Split Pay for Direct Debit ACH & EFT Standalone Credit requests
* adds unit tests for code modified to enable split pay for Direct Debit ACH & EFT Standalone Credit requests

### CODE REVIEW

File: source/Paysafe/DirectDebit/ShippingDetails.php [hunk 1]
```diff
@@ -30,7 +30,7 @@
  * @property string $country
  * @property string $zip
  */
-class ShippingDetails
+class ShippingDetails extends \Paysafe\JSONObject
```
Comment (PayLeaseBrian) [inline source/Paysafe/DirectDebit/ShippingDetails.php:33]: this missing `extends` was revealed by unit tests

File: source/Paysafe/DirectDebit/StandaloneCredits.php [hunk 2]
```diff
@@ -17,74 +17,78 @@
...
-                    if ($link->rel == $linkName) {
-                        return $link;
-                    }
+    /**
+     * @param string $linkName
+     * @return \Paysafe\Link
+     * @throws \Paysafe\PaysafeException
+     */
+    public function getLink( $linkName ) {
+        if (!empty($this->links)) {
```
Comment (PayLeaseBrian) [inline source/Paysafe/DirectDebit/StandaloneCredits.php:85]: Had a bit of a bug here. Had to change `$this->link` to `$this->links`. Also revealed by unit tests.

File: source/Paysafe/DirectDebit/StandaloneCredits.php [hunk 3]
```diff
@@ -17,74 +17,78 @@
...
+            foreach ($this->links as $link) {
+                if ($link->rel == $linkName) {
+                    return $link;
                 }
             }
-            throw new PaysafeException("Link $linkName not found in purchase.");
         }
-
+        throw new PaysafeException("Link $linkName not found in stand alone credit.");
```
Comment (PayLeaseBrian) [inline source/Paysafe/DirectDebit/StandaloneCredits.php:92]: Looks like our exception message was copy/pasted. I updated to reflect the correct context.

File: tests/paysafe/AccountManagement/TransferTest.php [hunk 4]
```diff
@@ -194,7 +194,7 @@ public function testAllFieldsValidValues()
          * This may seem like a trivial test, but behind the scenes toJson triggers data validation. Bad data will
          * result in an exception.
          * Not only does this test ensure the proper operation of the json encoding in JSONObject, but it validates
-         * our understanding of the data requirements in Authorization
+         * our understanding of the data requirements in Transfer
```
Comment (PayLeaseBrian) [inline tests/paysafe/AccountManagement/TransferTest.php:197]: just cleaning up an old comment I noticed

File: tests/paysafe/CardPayments/RefundTest.php [hunk 5]
```diff
@@ -183,7 +183,7 @@ public function testAllFieldsValueValues()
          * This may seem like a trivial test, but behind the scenes toJson triggers data validation. Bad data will
          * result in an exception.
          * Not only does this test ensure the proper operation of the json encoding in JSONObject, but it validates
-         * our understanding of the data requirements in Authorization
+         * our understanding of the data requirements in Refund
```
Comment (PayLeaseBrian) [inline tests/paysafe/CardPayments/RefundTest.php:186]: just cleaning up an old comment I noticed

File: tests/paysafe/DirectDebitServiceStandaloneCreditsTest.php [hunk 6]
```diff
@@ -0,0 +1,515 @@
...
+        $this->mock_api_client->method('getAccount')->willReturn('bogus_account_num');
+    }
+
+    /**
+     * This is a bad test as it simply confirms current undesirable behavior. If no type is specified in the
+     * StandaloneCredits object (ach, eft, etc), then a PHP Error is generated. Ideally, the code would gracefully
+     * handle this situation.
+     * See: https://github.com/paysafegroup/paysafe_sdk_php/issues/13
+     */
+    public function testMissingBankInfo()
```
Comment (PayLeaseBrian) [inline tests/paysafe/DirectDebitServiceStandaloneCreditsTest.php:37]: This test will need to change when/if #13 is addressed.

**Review by jbPayLease — APPROVED** [review]

**Pull request merged**